src/GafferImage/Display.cpp

```diff
@@ -228,23 +225,59 @@ class GafferDisplayDriver : public IECoreImage::DisplayDriver
 			return true;
 		}
 
-		ConstFloatVectorDataPtr channelData( const Imath::V2i &tileOrigin, const std::string &channelName )
+		ConstFloatVectorDataPtr channelData( const Imath::V2i &tileOrigin, const std::string &channelName, int dataCount )
 		{
 			vector<string>::const_iterator cIt = find( channelNames().begin(), channelNames().end(), channelName );
 			if( cIt == channelNames().end() )
 			{
 				return ImagePlug::blackTile();
 			}
 
-			ConstFloatVectorDataPtr tile = getTile( tileOrigin, cIt - channelNames().begin() );
-			if( tile )
+			Tile *tile = getTile( tileOrigin, cIt - channelNames().begin() );
+			if( !tile )
+			{
+				return ImagePlug::blackTile();
+			}
+
+			int previousDataCount;
+
+			{
+				tbb::spin_rw_mutex::scoped_lock tileLock( tile->mutex, false /* read */ );
```

Comment:
I find `false /* read */` really ambiguous : are we setting a parameter called read to false? I would prefer `/* write = */ false` for clarity, and eagerly await keyword arguments in C++ one day.

Reply:
Addressed.